### PR TITLE
[W5][T12-3]Melvin Vito

### PR DESCRIPTION
**Pull request by melpulomas**
Added a change command which allows users to change the phone numbers of persons by their index.
Also updated the user guide, as well as I/O and JUnit tests.

### CODE REVIEW

**Review by kevvvinn — COMMENTED** [review]
Great and useful command! Maybe include an option to change other fields such as email address and name!

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -3,15 +3,12 @@
 import static seedu.addressbook.common.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
 import static seedu.addressbook.common.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.*;
 import java.util.regex.Matcher;
```
Comment (nhs-work) [inline src/seedu/addressbook/parser/Parser.java:7]: Imported classes should always be listed explicitly.

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -103,6 +104,24 @@ public Command parseCommand(String userInput) {
         }
     }
 
+    /**
+     * Parses arguments in the context of the add person command.
```
Comment (nhs-work) [inline src/seedu/addressbook/parser/Parser.java:108]: Don't forget to update the comments to reflect different commands. It should be "in the context of the change person command".

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -103,6 +104,24 @@ public Command parseCommand(String userInput) {
...
+     * @return the change Command
+     */
+    private Command prepareChange(String args) {
+        try {
+            Scanner sc = new Scanner(System.in);
+            System.out.print("|| Enter new Number: ");
+            String newNumber = sc.nextLine();
+            final int targetIndex = parseArgsAsDisplayedIndex(args);
+            return new ChangeCommand(targetIndex, newNumber);
+        }   catch (ParseException pe) {
```
Comment (nhs-work) [inline src/seedu/addressbook/parser/Parser.java:120]: There should only be 1 space between the close brace and the catch statement.

File: test/java/seedu/addressbook/commands/ChangeCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        assertEquals(expectedAddressBook.getAllPersons(), actualAddressBook.getAllPersons());
+    }
+
+    /**
+     * Asserts that the index is not valid for the given display list.
+     */
+    private void assertChangeFailsDueToInvalidIndex(int invalidVisibleIndex, AddressBook addressBook,
+                                                      List<ReadOnlyPerson> displayList) {
+
+        String expectedMessage = Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
```
Comment (nhs-work) [inline test/java/seedu/addressbook/commands/ChangeCommandTest.java:109]: Unnecessary blank lines here as well.

File: src/seedu/addressbook/commands/ChangeCommand.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    public ChangeCommand(int targetVisibleIndex, String newNumber ) {
+        super(targetVisibleIndex);
+        this.newNumber = newNumber;
+    }
+
+
+
+
+    @Override
```
Comment (nhs-work) [inline src/seedu/addressbook/commands/ChangeCommand.java:37]: Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

**Review by nhs-work — APPROVED** [review]
Great job on your first PR! Please close the PR after you have read my comments.

**Pull request closed without merging**